# **1) Temporal Analysis**

In [0]:
-- 1) Temporal Analysis
-- yearly reserved vulnerability counts
-- All data from 2024?

SELECT
  YEAR(date_reserved) AS year,
  COUNT(*) AS vulnerability_count
FROM workspace.cve_silver.core
WHERE date_reserved IS NOT NULL
GROUP BY year
ORDER BY year

year vulnerability_count 2023 1519 2024 35595 2025 1613

Databricks visualization. Run in Databricks to view.

In [0]:
-- yearly published vulnerability counts
-- All data from 2024?

SELECT
  YEAR(date_published) AS year,
  COUNT(*) AS vulnerability_count_published
FROM workspace.cve_silver.core
WHERE date_published IS NOT NULL
GROUP BY year
ORDER BY year

year vulnerability_count_published 2024 32924 2025 5370

Databricks visualization. Run in Databricks to view.

In [0]:
-- Publication latency analysis: difference between dateReserved and datePublished

SELECT
  cve_id,
  date_reserved,
  date_published,
  DATEDIFF(
    date_published,
    date_reserved
  ) AS publication_latency_days
FROM workspace.cve_silver.core
WHERE date_reserved IS NOT NULL
  AND date_published IS NOT NULL
ORDER BY publication_latency_days DESC


cve_id date_reserved date_published publication_latency_days CVE-2024-0028 2023-11-16T22:58:45.676Z 2025-09-05T16:10:01.094Z 659 CVE-2024-25621 2024-02-08T22:26:33.511Z 2025-11-06T18:36:21.566Z 637 CVE-2024-21927 2024-01-03T16:43:09.233Z 2025-09-23T21:33:54.121Z 629 CVE-2024-21935 2024-01-03T16:43:14.976Z 2025-09-23T21:38:22.057Z 629 CVE-2024-21947 2024-01-03T16:43:21.322Z 2025-09-06T17:10:47.951Z 612 CVE-2024-21970 2024-01-03T16:43:28.699Z 2025-09-06T17:20:19.749Z 612 CVE-2024-21977 2024-01-03T16:43:30.196Z 2025-09-05T12:58:39.312Z 611 CVE-2024-26008 2024-02-14T09:18:43.245Z 2025-10-14T15:23:04.753Z 608 CVE-2024-25011 2024-02-02T21:33:13.076Z 2025-09-18T11:38:18.371Z 594 CVE-2024-31573 2024-04-05T00:00:00.000Z 2025-10-17T00:00:00.000Z 560 CVE-2024-26009 2024-02-14T09:18:43.245Z 2025-08-12T18:59:47.462Z 545 CVE-2024-33507 2024-04-23T14:18:29.830Z 2025-10-14T15:23:42.194Z 539 CVE-2024-28988 2024-03-13T20:27:09.782Z 2025-09-01T21:18:58.626Z 537 CVE-2024-25176 2024-02-07T00:00:00.000Z 2025-07-07T00:00:00.000Z 516 CVE-2024-25177 2024-02-07T00:00:00.000Z 2025-07-07T00:00:00.000Z 516 CVE-2024-25178 2024-02-07T00:00:00.000Z 2025-07-07T00:00:00.000Z 516 CVE-2024-22330 2024-01-08T23:42:07.732Z 2025-06-06T01:08:52.194Z 515 CVE-2024-26292 2024-02-16T16:12:43.383Z 2025-07-14T08:19:58.940Z 514 CVE-2024-24915 2024-02-01T15:19:26.278Z 2025-06-29T12:02:41.126Z 514 CVE-2024-26291 2024-02-16T16:12:43.383Z 2025-07-14T08:12:16.139Z 514 CVE-2024-26293 2024-02-16T16:12:43.383Z 2025-07-14T09:18:18.045Z 514 CVE-2024-32444 2024-04-12T14:58:00.298Z 2025-09-03T06:55:35.317Z 509 CVE-2024-27779 2024-02-26T14:46:31.334Z 2025-07-18T07:58:23.943Z 508 CVE-2024-22653 2024-01-11T00:00:00.000Z 2025-05-29T00:00:00.000Z 504 CVE-2024-4598 2024-05-07T06:40:12.013Z 2025-09-23T10:39:16.195Z 504 CVE-2024-24916 2024-02-01T15:19:26.278Z 2025-06-19T13:17:39.651Z 504 CVE-2024-22654 2024-01-11T00:00:00.000Z 2025-05-29T00:00:00.000Z 504 CVE-2024-32589 2024-04-15T10:18:19.798Z 2025-08-31T03:46:56.764Z 503 CVE-2024-32832 2024-04-18T11:08:33.814Z 2025-08-31T03:47:28.414Z 500 CVE-2024-23589 2024-01-18T07:30:10.662Z 2025-05-30T15:36:13.985Z 498 CVE-2024-0249 2024-01-05T14:11:46.692Z 2025-05-15T20:09:29.403Z 496 CVE-2024-21960 2024-01-03T16:43:26.978Z 2025-05-13T16:54:53.805Z 496 CVE-2024-5200 2024-05-22T13:44:22.851Z 2025-09-29T06:00:10.942Z 495 CVE-2024-23337 2024-01-15T15:19:19.443Z 2025-05-21T14:34:51.007Z 492 CVE-2024-1244 2024-02-06T00:37:58.090Z 2025-06-11T02:59:06.240Z 491 CVE-2024-1243 2024-02-06T00:37:55.742Z 2025-06-11T01:15:13.116Z 491 CVE-2024-34598 2024-05-07T04:43:27.829Z 2025-09-04T10:55:34.535Z 485 CVE-2024-27685 2024-02-26T00:00:00.000Z 2025-06-25T00:00:00.000Z 485 CVE-2024-32640 2024-04-16T14:15:26.874Z 2025-08-11T20:38:56.268Z 482 CVE-2024-0852 2024-01-24T08:26:22.019Z 2025-05-15T20:09:31.811Z 477 CVE-2024-23815 2024-01-22T17:44:56.763Z 2025-05-13T09:38:11.730Z 477 CVE-2024-1440 2024-02-12T09:53:51.193Z 2025-06-02T16:51:16.948Z 476 CVE-2024-0970 2024-01-26T19:42:29.765Z 2025-05-15T20:09:32.658Z 475 CVE-2024-25010 2024-02-02T21:33:13.076Z 2025-05-22T10:14:00.320Z 475 CVE-2024-25573 2024-02-29T23:52:30.507Z 2025-06-15T15:25:38.540Z 472 CVE-2024-36346 2024-05-23T19:44:47.201Z 2025-09-06T17:43:14.113Z 471 CVE-2024-36352 2024-05-23T19:44:50.000Z 2025-09-06T17:54:57.932Z 471 CVE-2024-36342 2024-05-23T19:44:47.200Z 2025-09-06T17:42:00.232Z 471 CVE-2024-36326 2024-05-23T19:44:40.301Z 2025-09-06T17:25:46.065Z 471 CVE-2024-36354 2024-05-23T19:44:50.000Z 2025-09-06T18:06:43.084Z 471 CVE-2024-35203 2024-05-12T00:00:00.000Z 2025-08-26T00:00:00.000Z 471 CVE-2024-22351 2024-01-08T23:42:25.451Z 2025-04-23T22:15:49.142Z 471 CVE-2024-36331 2024-05-23T19:44:44.387Z 2025-09-06T17:29:38.215Z 471 CVE-2024-24780 2024-01-30T10:43:03.969Z 2025-05-14T10:42:20.580Z 470 CVE-2024-22036 2024-01-04T12:38:34.025Z 2025-04-16T08:37:54.218Z 468 CVE-2024-22314 2024-01-08T23:41:52.508Z 2025-04-16T16:17:54.793Z 464 CVE-2024-32124 2024-04-11T12:09:46.571Z 2025-07-18T08:08:21.544Z 463 CVE-2024-32323 

Databricks visualization. Run in Databricks to view.

Databricks visualization. Run in Databricks to view.

In [0]:
-- Seasonal vulnerability patterns
-- Monthly counts of reserved and published vulnerabilities

SELECT
  month,
  SUM(vulnerability_count_reserved) AS vulnerability_count_reserved,
  SUM(vulnerability_count_published) AS vulnerability_count_published
FROM (
  SELECT
    MONTH(date_reserved) AS month,
    COUNT(*) AS vulnerability_count_reserved,
    0 AS vulnerability_count_published
  FROM workspace.cve_silver.core
  WHERE date_reserved IS NOT NULL
  GROUP BY MONTH(date_reserved)

  UNION ALL

  SELECT
    MONTH(date_published) AS month,
    0 AS vulnerability_count_reserved,
    COUNT(*) AS vulnerability_count_published
  FROM workspace.cve_silver.core
  WHERE date_published IS NOT NULL
  GROUP BY MONTH(date_published)
) AS combined
GROUP BY month
ORDER BY month


month vulnerability_count_reserved vulnerability_count_published 1 4116 3320 2 3344 2834 3 3268 3573 4 3086 3484 5 3160 3781 6 2968 2835 7 2552 2991 8 3114 2794 9 2479 2483 10 3782 3410 11 3435 3767 12 3423 3022

Databricks visualization. Run in Databricks to view.

Findings: 

Though, most CVEs are reserved and published in their corresponding year, there is some minor overlap between surrounding years. While most CVEs are published with some immediacy (~10 days of latency), some are published much later, extending out past one year, with one CVE being published nearly two years after reservation. Further analysis could be done to analyze what companies/cvss priorities correlate to these longer latencies.

# **2) Risk Distribution Analysis**

In [0]:
-- 2) Risk Distribution Analysis
-- CVSS score bucketing and risk severity trends over time

SELECT
  YEAR(date_published) AS year,
  MONTH(date_published) AS month,
  CASE
    WHEN cvss_score >= 9 THEN 'Critical (9+)'
    WHEN cvss_score >= 7 THEN 'High (7 - 9)'
    WHEN cvss_score >= 4 THEN 'Medium (4 - 7)'
    WHEN cvss_score >= 0 THEN 'Low (0 - 4)'
    ELSE 'Unknown'
  END AS severity_bucket,
  COUNT(*) AS vulnerability_count
FROM workspace.cve_silver.core
WHERE date_published IS NOT NULL
GROUP BY year, month, severity_bucket
ORDER BY year, month, severity_bucket;



year month severity_bucket vulnerability_count 2024 1 Critical (9+) 48 2024 1 High (7 - 9) 282 2024 1 Low (0 - 4) 91 2024 1 Medium (4 - 7) 445 2024 1 Unknown 268 2024 2 Critical (9+) 100 2024 2 High (7 - 9) 372 2024 2 Low (0 - 4) 80 2024 2 Medium (4 - 7) 657 2024 2 Unknown 560 2024 3 Critical (9+) 109 2024 3 High (7 - 9) 644 2024 3 Low (0 - 4) 95 2024 3 Medium (4 - 7) 1005 2024 3 Unknown 763 2024 4 Critical (9+) 128 2024 4 High (7 - 9) 734 2024 4 Low (0 - 4) 146 2024 4 Medium (4 - 7) 1148 2024 4 Unknown 1062 2024 5 Critical (9+) 147 2024 5 High (7 - 9) 540 2024 5 Low (0 - 4) 72 2024 5 Medium (4 - 7) 1213 2024 5 Unknown 1376 2024 6 Critical (9+) 169 2024 6 High (7 - 9) 532 2024 6 Low (0 - 4) 63 2024 6 Medium (4 - 7) 1124 2024 6 Unknown 819 2024 7 Critical (9+) 129 2024 7 High (7 - 9) 615 2024 7 Low (0 - 4) 69 2024 7 Medium (4 - 7) 970 2024 7 Unknown 1094 2024 8 Critical (9+) 153 2024 8 High (7 - 9) 590 2024 8 Low (0 - 4) 42 2024 8 Medium (4 - 7) 1034 2024 8 Unknown 873 2024 9 Critical (9+) 165 2024 9 High (7 - 9) 477 2024 9 Low (0 - 4) 54 2024 9 Medium (4 - 7) 799 2024 9 Unknown 913 2024 10 Critical (9+) 199 2024 10 High (7 - 9) 809 2024 10 Low (0 - 4) 89 2024 10 Medium (4 - 7) 1237 2024 10 Unknown 1039 2024 11 Critical (9+) 224 2024 11 High (7 - 9) 981 2024 11 Low (0 - 4) 129 2024 11 Medium (4 - 7) 1454 2024 11 Unknown 972 2024 12 Critical (9+) 170 2024 12 High (7 - 9) 778 2024 12 Low (0 - 4) 54 2024 12 Medium (4 - 7) 1211 2024 12 Unknown 809 2025 1 Critical (9+) 138 2025 1 High (7 - 9) 303 2025 1 Low (0 - 4) 50 2025 1 Medium (4 - 7) 777 2025 1 Unknown 918 2025 2 Critical (9+) 35 2025 2 High (7 - 9) 204 2025 2 Low (0 - 4) 45 2025 2 Medium (4 - 7) 411 2025 2 Unknown 370 2025 3 Critical (9+) 70 2025 3 High (7 - 9) 318 2025 3 Low (0 - 4) 28 2025 3 Medium (4 - 7) 311 2025 3 Unknown 230 2025 4 Critical (9+) 13 2025 4 High (7 - 9) 52 2025 4 Low (0 - 4) 19 2025 4 Medium (4 - 7) 101 2025 4 Unknown 81 2025 5 Critical (9+) 9 2025 5 High (7 - 9) 69 2025 5 Low (0 - 4) 8 2025 5 Medium (4 - 7) 102 2025 5 Unknown 245 2025 6 Critical (9+) 18 2025 6 High (7 - 9) 29 2025 6 Low (0 - 4) 5 2025 6 Medium (4 - 7) 50 2025 6 Unknown 26 2025 7 Critical (9+) 3 2025 7 High (7 - 9) 26 2025 7 Low (0 - 4) 10 2025 7 Medium (4 - 7) 40 2025 7 Unknown 35 2025 8 Critical (9+) 8 2025 8 High (7 - 9) 12 2025 8 Low (0 - 4) 8 2025 8 Medium (4 - 7) 23 2025 8 Unknown 51 2025 9 Critical (9+) 6 2025 9 High (7 - 9) 16 2025 9 Low (0 - 4) 2 2025 9 Medium (4 - 7) 25 2025 9 Unknown 26 2025 10 Critical (9+) 5 2025 10 High (7 - 9) 14 2025 10 Medium (4 - 7) 13 2025 10 Unknown 5 2025 11 Critical (9+) 1 2025 11 High (7 - 9) 1 2025 11 Medium (4 - 7) 3 2025 11 Unknown 2

Databricks visualization. Run in Databricks to view.

In [0]:
SELECT
  CASE
    WHEN cvss_score >= 9 THEN 'Critical (9+)'
    WHEN cvss_score >= 7 THEN 'High (7 - 9)'
    WHEN cvss_score >= 4 THEN 'Medium (4 - 7)'
    WHEN cvss_score >= 0 THEN 'Low (0 - 4)'
    ELSE 'Unknown'
  END AS severity_bucket,
  COUNT(*) AS vulnerability_count
FROM workspace.cve_silver.core
GROUP BY severity_bucket
ORDER BY severity_bucket;

severity_bucket vulnerability_count Critical (9+) 2047 High (7 - 9) 8398 Low (0 - 4) 1159 Medium (4 - 7) 14153 Unknown 12970

Databricks visualization. Run in Databricks to view.

Findings:

The prevalence of each cvss severity does not appear to change with time. Each month shares a similar pattern of ~35% Unknown, ~5% Low, ~35% Medium, ~20% High, and ~5% Critical. Interestingly, the prevalence of each severity almost follows a pyramid pattern (where the highest prevalence is associated with the lowest severity, and the lowest prevalence with the highest severity), however, the low severity takes up an extremely small portion of the published CVEs. This may be because many CVEs are such low priority that they are not even rated on the cvss scale, resulting in the large portion in the "Unknown" category. More research into cvss ratings will be needed to confirm/deny this idea.

# **3) Vendor Intelligence**

In [0]:
-- 3) Vendor Intelligence
-- Top 25 vendors by vulnerability count

SELECT
  vendor,
  COUNT(*) AS vulnerability_count
FROM workspace.cve_silver.affected
GROUP BY vendor
ORDER BY vulnerability_count DESC
LIMIT 25

vendor vulnerability_count Microsoft 13161 Unknown 8770 Linux 6152 Brother Industries, Ltd 4427 Red Hat 3913 Siemens 2540 Apple 1692 Lenovo 929 Adobe 751 Autodesk 743 NEC Corporation 717 Gigabyte 648 Google 630 D-Link 616 IBM 570 SourceCodester 558 AMD 546 PHOENIX CONTACT 512 Mozilla 498 Oracle Corporation 377 Cisco 363 Dell 316 Fortinet 268 Apache Software Foundation 263 Qualcomm, Inc. 262

Databricks visualization. Run in Databricks to view.

In [0]:
-- Market concentration analysis: percentage of vulnerabilities attributed to top vendors

SELECT
  vendor,
  COUNT(*) AS vulnerability_count,
  ROUND(100.0 * COUNT(*) / SUM(COUNT(*)) OVER (), 2) AS vulnerability_pct
FROM workspace.cve_silver.affected
GROUP BY vendor
ORDER BY vulnerability_count DESC
LIMIT 25

vendor vulnerability_count vulnerability_pct Microsoft 13161 17.53 Unknown 8770 11.68 Linux 6152 8.20 Brother Industries, Ltd 4427 5.90 Red Hat 3913 5.21 Siemens 2540 3.38 Apple 1692 2.25 Lenovo 929 1.24 Adobe 751 1.00 Autodesk 743 0.99 NEC Corporation 717 0.96 Gigabyte 648 0.86 Google 630 0.84 D-Link 616 0.82 IBM 570 0.76 SourceCodester 558 0.74 AMD 546 0.73 PHOENIX CONTACT 512 0.68 Mozilla 498 0.66 Oracle Corporation 377 0.50 Cisco 363 0.48 Dell 316 0.42 Fortinet 268 0.36 Apache Software Foundation 263 0.35 Qualcomm, Inc. 262 0.35

Databricks visualization. Run in Databricks to view.

In [0]:
-- Vendor-specific risk profiles: CVSS severity distribution and vulnerability counts per vendor
-- Limit to top 25 vendors
WITH top_vendors AS (
  SELECT
    a.vendor,
    COUNT(*) AS total_vulns
  FROM workspace.cve_silver.affected a
  JOIN workspace.cve_silver.core c
    ON a.cve_id = c.cve_id
  GROUP BY a.vendor
  ORDER BY total_vulns DESC
  LIMIT 25
)


SELECT
  a.vendor as vendor,
  COUNT(*) AS vulnerability_count,
  AVG(c.cvss_score) AS avg_cvss_score,
  CASE
    WHEN c.cvss_score >= 9 THEN 'Critical (9+)'
    WHEN c.cvss_score >= 7 THEN 'High (7 - 9)'
    WHEN c.cvss_score >= 4 THEN 'Medium (4 - 7)'
    WHEN c.cvss_score >= 0 THEN 'Low (0 - 4)'
    ELSE 'Unknown'
  END AS severity_bucket
FROM workspace.cve_silver.affected a
JOIN workspace.cve_silver.core c
  ON a.cve_id = c.cve_id
JOIN top_vendors v
  ON a.vendor = v.vendor
GROUP BY a.vendor, severity_bucket
ORDER BY vulnerability_count DESC, severity_bucket


vendor vulnerability_count avg_cvss_score severity_bucket Microsoft 9516 8.017969735183756 High (7 - 9) Unknown 7578 null Unknown Linux 6138 null Unknown Microsoft 3375 6.105214814814762 Medium (4 - 7) Brother Industries, Ltd 2355 5.705732484076205 Medium (4 - 7) Red Hat 2117 7.682239017477535 High (7 - 9) Apple 1692 null Unknown Red Hat 1660 5.686084337349303 Medium (4 - 7) Brother Industries, Ltd 1383 7.374837310195196 High (7 - 9) Siemens 1338 5.38505231689092 Medium (4 - 7) Siemens 935 7.766524064171133 High (7 - 9) Autodesk 742 7.8000000000001055 High (7 - 9) Unknown 723 5.7118948824343265 Medium (4 - 7) NEC Corporation 710 null Unknown Brother Industries, Ltd 689 9.800000000000118 Critical (9+) Gigabyte 648 null Unknown Google 603 null Unknown Lenovo 589 7.221561969439656 High (7 - 9) Mozilla 498 null Unknown Adobe 494 5.389676113360353 Medium (4 - 7) SourceCodester 470 5.6597872340425655 Medium (4 - 7) IBM 429 5.500932400932409 Medium (4 - 7) PHOENIX CONTACT 412 8.377669902912645 High (7 - 9) D-Link 396 5.39494949494947 Medium (4 - 7) Unknown 332 7.779518072289157 High (7 - 9) Lenovo 326 5.601226993865022 Medium (4 - 7) Microsoft 266 9.654511278195505 Critical (9+) AMD 236 8.069491525423738 High (7 - 9) Oracle Corporation 236 5.4427966101694905 Medium (4 - 7) Cisco 234 5.691025641025631 Medium (4 - 7) Adobe 232 7.789224137931007 High (7 - 9) AMD 214 5.74719626168226 Medium (4 - 7) Apache Software Foundation 212 null Unknown D-Link 197 8.625888324873113 High (7 - 9) Qualcomm, Inc. 189 7.848148148148138 High (7 - 9) Fortinet 155 5.589032258064514 Medium (4 - 7) Siemens 138 9.73768115942028 Critical (9+) Siemens 129 3.0720930232558117 Low (0 - 4) Dell 123 5.721138211382111 Medium (4 - 7) Dell 110 7.758181818181813 High (7 - 9) Cisco 107 8.099065420560754 High (7 - 9) PHOENIX CONTACT 96 5.441666666666669 Medium (4 - 7) AMD 95 3.6547368421052697 Low (0 - 4) IBM 94 7.849999999999997 High (7 - 9) Red Hat 88 3.285227272727272 Low (0 - 4) Oracle Corporation 82 7.851219512195124 High (7 - 9) Unknown 70 3.0985714285714296 Low (0 - 4) Qualcomm, Inc. 70 6.432857142857144 Medium (4 - 7) Unknown 67 9.708955223880594 Critical (9+) Fortinet 63 7.574603174603175 High (7 - 9) Oracle Corporation 51 3.0823529411764703 Low (0 - 4) SourceCodester 50 3.338 Low (0 - 4) Dell 49 null Unknown Red Hat 48 9.735416666666671 Critical (9+) SourceCodester 38 7.300000000000005 High (7 - 9) Fortinet 38 3.0842105263157893 Low (0 - 4) IBM 31 3.148387096774193 Low (0 - 4) Dell 25 3.1719999999999993 Low (0 - 4) D-Link 23 9.573913043478266 Critical (9+) Apache Software Foundation 21 5.447619047619047 Medium (4 - 7) Cisco 20 9.830000000000002 Critical (9+) Apache Software Foundation 19 8.294736842105262 High (7 - 9) IBM 16 9.48125 Critical (9+) Adobe 14 9.442857142857141 Critical (9+) Google 13 6.115384615384615 Medium (4 - 7) Fortinet 12 9.674999999999997 Critical (9+) Lenovo 12 null Unknown Adobe 11 3.1727272727272733 Low (0 - 4) Google 10 8.440000000000001 High (7 - 9) Linux 10 5.0 Medium (4 - 7) Dell 9 9.666666666666666 Critical (9+) Oracle Corporation 8 9.6375 Critical (9+) Apache Software Foundation 7 9.628571428571428 Critical (9+) PHOENIX CONTACT 4 9.8 Critical (9+) Google 4 10.0 Critical (9+) NEC Corporation 4 7.2 High (7 - 9) Apache Software Foundation 4 2.6 Low (0 - 4) Qualcomm, Inc. 3 9.8 Critical (9+) Microsoft 3 3.233333333333333 Low (0 - 4) NEC Corporation 3 4.3 Medium (4 - 7) Linux 2 7.8 High (7 - 9) Lenovo 2 2.4 Low (0 - 4) Linux 2 3.4 Low (0 - 4) Cisco 1 3.8 Low (0 - 4) Autodesk 1 5.5 Medium (4 - 7) AMD 1 null Unknown Microsoft 1 null Unknown Cisco 1 null Unknown

Databricks visualization. Run in Databricks to view.

Databricks visualization. Run in Databricks to view.

In [0]:
-- average cvss score by vendor
-- Limit top 25 vendors

SELECT
  a.vendor as vendor,
  COUNT(*) AS vulnerability_count,
  AVG(c.cvss_score) AS avg_cvss_score
FROM workspace.cve_silver.affected a
JOIN workspace.cve_silver.core c
  ON a.cve_id = c.cve_id
WHERE c.cvss_score IS NOT NULL
GROUP BY vendor
ORDER BY vulnerability_count DESC, avg_cvss_score DESC
LIMIT 25


vendor vulnerability_count avg_cvss_score Microsoft 13160 7.5594148936179995 Brother Industries, Ltd 4427 6.864377682403008 Red Hat 3913 6.761717352415011 Siemens 2540 6.380708661417283 Unknown 1192 6.358976510067144 Lenovo 917 6.635005452562651 Adobe 751 6.174034620506047 Autodesk 743 7.796904441453672 D-Link 616 6.584253246753307 IBM 570 5.872105263157919 SourceCodester 558 5.563440860215079 AMD 545 6.38807339449546 PHOENIX CONTACT 512 7.838281250000007 Oracle Corporation 377 5.736339522546424 Cisco 362 6.626243093922649 Fortinet 268 5.883582089552237 Dell 267 6.454681647940072 Qualcomm, Inc. 262 7.492366412213734 Tenda 242 8.14297520661155 FUJIFILM Business Innovation 238 6.189915966386537 code-projects 238 5.893697478991582 Ivanti 232 7.8491379310344715 Huawei 232 6.118965517241386 Samsung Mobile 221 5.436199095022622 ABB 216 7.624537037037028

Databricks visualization. Run in Databricks to view.

Findings:

As expected, most published CVEs are associated with widely recognizable names in the technology industries, such as Microsoft, Apple, Google, etc. Interestingly, some of these companies (Apple, Gigabyte, Google, Linux, Mozilla, and NEC) have no/almost no CVEs with cvss scores. It seems obvious that these companies would have at least a few rateable CVEs published, suggesting that there is some reason that they are not scored, be it data privacy or maintaining a certain public image. Another interesting finding is the prevalence of critical priority CVEs published for Brother Industries (15.6%, 589 CVes). This percentage and quantity is much higher than other companies listed, with the next highest percentage being from Cisco (5.51%, 20 CVEs) and the next highest quantity being from Microsoft (2.02%, 266 CVEs). A cursory search shows Brother Industries is a manufacturer of sewing machines and printers, leading to the question of why these products are resulting in such a high quantity of critical priority CVEs. It could possibly be due to the security issues associated with wireless connection to printers that are used for secured data, but an analysis of these CVEs may prove interesting.